src/components/Explorer.vue

```diff
@@ -749,15 +762,79 @@ export default {
     await this.determineFilterCountsAndSubmit();
     await this.fetchPage();
   },
+  async postData(url = "", data = {}) {
+    // Default options are marked with *
+    const response = await fetch(url, {
+      method: "POST", // *GET, POST, PUT, DELETE, etc.
+       headers: {
+        "Content-Type": "application/json",
+        // 'Content-Type': 'application/x-www-form-urlencoded',
+       },
+       body: JSON.stringify(data), // body data type must match "Content-Type" header
+    });
+    return response.json(); // parses JSON response into native JavaScript objects
+  },
   // Browser only
   async mounted() {
+    if ("geolocation" in navigator) {
+      navigator.geolocation.getCurrentPosition(async (position) => {
+        console.log("POSITION", position)
+        var data = {latitude:position.coords.latitude, longitude:position.coords.longitude};
+         //get zipcode by lng/lat
+        //get state by lng / lat
+        const response = await fetch("/get-zip", {
+          method: "POST", // *GET, POST, PUT, DELETE, etc.
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(data), // body data type must match "Content-Type" header
+        });
+        var json = await response.json();
+         console.log('RESPONSE', json)
+        this.zipCode = json.code;
+        this.filterValues["States"] = [json.state]  
+        this.taisplayLocation = `${json.city}, ${json.state}`;
+      });
+    }else{
+      console.log("Location not supported")
+    }
+   
     await this.determineFilterCountsAndSubmit();
     await this.fetchSelectedIfNeeded();
   },
   destroy() {
     document.body.removeEventListener('click', this.bodyClick);
   },
   methods: {
+    async setLocation(){
+      this.zipCode = prompt("Please enter your zipcode")
```

Comment:
`prompt` isn't the best user experience but it can be a reasonable iterative step...

However, what if they cancel rather than entering a zip code?

Reply:
If they cancel then the zipcode is null and everything is as it was.  Prompt is much pretty today in modern browsers than it once was in < IE 10.

Reply:
Changed behavior.